site/src/cms/widgets/netlify-cms-media-library-uploadcare-custom/index.js

```diff
@@ -0,0 +1,244 @@
+  )
+}
+
+/**
+ * Convert a url or array/List of urls to Uploadcare file objects wrapped in
+ * promises, or Uploadcare groups when possible. Output is wrapped in a promise
+ * because the value we're returning may be a promise that we created.
+ */
+function getFiles(value) {
+  if (Array.isArray(value) || Iterable.isIterable(value)) {
+    const arr = Array.isArray(value) ? value : value.toJS()
+    return isFileGroup(arr)
+      ? getFileGroup(arr)
+      : Promise.all(arr.map(val => getFile(val)))
+  }
+  return value && typeof value === "string" ? getFile(value) : null
+}
+
+/**
+ * Convert a single url to an Uploadcare file object wrapped in a promise-like
+ * object. Group urls that get passed here were not a part of a complete and
+ * untouched group, so they'll be uploaded as new images (only way to do it).
+ */
+function getFile(url) {
+  const groupPattern = /~\d+\/nth\/\d+\//
+  const uploaded = url.startsWith(CDN_BASE_URL) && !groupPattern.test(url)
+  return uploadcare.fileFrom(uploaded ? "uploaded" : "url", url)
+}
+
+/**
+ * Open the standalone dialog. A single instance is created and destroyed for
+ * each use.
+ */
+async function openDialog({ files, config, handleInsert, settings = {} }) {
+  if (
+    settings.defaultOperations &&
+    !settings.defaultOperations.startsWith("/")
+  ) {
+    console.warn(
+      "Uploadcare default operations should start with `/`. Example: `/preview/-/resize/100x100/image.png`"
+    )
+  }
+
+  const buildUrl = fileInfo => {
+    const { cdnUrl, name, isImage } = fileInfo
+
+    let url =
+      isImage && settings.defaultOperations
+        ? `${cdnUrl}-${settings.defaultOperations}`
+        : cdnUrl
+    const filenameDefined = !url.endsWith("/")
+
+    if (!filenameDefined && settings.autoFilename) {
+      url = url + name
+    }
+
+    return url
+  }
+
+  if (config["useSecureUpload"]) {
+    const { signature, expiry } = await getSignature()
+
+    config["secureSignature"] = signature
+    config["secureExpire"] = expiry
+  }
+
+  const sendToSThree = uuid => {
+    // TODO : move all this code into an API to protect the private key
+    const myHeaders = new Headers()
+    myHeaders.append("Content-Type", "application/x-www-form-urlencoded")
+
+    const urlencoded = new URLSearchParams()
+    urlencoded.append("uuid", uuid)
+
+    const requestOptions = {
+      method: "POST",
+      headers: myHeaders,
+      body: urlencoded,
+      redirect: "follow"
+    }
+
+    return fetch("/api/uploadcare/copy", requestOptions)
+      .then(res => res.json())
+      .then(json => {
+        return json.s3Url
+      })
+      .catch(err => console.error(err))
+  }
+
+  uploadcare.openDialog(files, config).done(({ promise, files }) => {
+    const isGroup = Boolean(files)
+
+    return promise().then(info => {
+      if (isGroup) {
+        return Promise.all(
+          files().map(promise => promise.then(fileInfo => buildUrl(fileInfo)))
+        ).then(urls => {
+          // TODO
```

Comment:
Remove. Point to an ONS Jira issue.

Reply:
https://worthit.jira.com/browse/ONS-800 done